### PR TITLE
Update the sample app to use a random nonce in the Sign in with Apple flow

### PR DESCRIPTION
**Pull request by rosalyntan**
Update the sample app to generate a random nonce for the Sign in with Apple flow, to minimize developer setup needed to test Sign in with Apple in the sample app.

Nonce generation same as sample code in https://firebase.google.com/docs/auth/ios/apple.

### CODE REVIEW

File: FirebaseAuth/Tests/Sample/Sample/MainViewController+Internal.h [hunk 1]
```diff
@@ -51,6 +51,8 @@ typedef void (^TestAutomationCallback)(NSError *_Nullable error);
 
 @property(nonatomic) NSURL *actionCodeContinueURL;
 
+@property(nonatomic) NSString *appleNonce;
```
Comment (renkelvin) [inline FirebaseAuth/Tests/Sample/Sample/MainViewController+Internal.h:54]: Use `copy` for `NSString` properties. go/objc-style#setters-copy-nsstrings

File: FirebaseAuth/Tests/Sample/Sample/MainViewController+OAuth.m [hunk 2]
```diff
@@ -321,11 +323,57 @@ - (void)signInYahooHeadfulLite {
 - (ASAuthorizationAppleIDRequest *)appleIDRequestWithState:(NSString *)state API_AVAILABLE(ios(13.0)) {
   ASAuthorizationAppleIDRequest *request = [[[ASAuthorizationAppleIDProvider alloc] init] createRequest];
   request.requestedScopes = @[ASAuthorizationScopeEmail, ASAuthorizationScopeFullName];
-  request.nonce = @"REPLACE_ME_WITH_YOUR_NONCE";
+  NSString *nonce = [self randomNonce:32];
```
Comment (renkelvin) [inline FirebaseAuth/Tests/Sample/Sample/MainViewController+OAuth.m:326]: Maybe rename it to `rawNonce` and `self.rawNonce`?

File: FirebaseAuth/Tests/Sample/Sample/MainViewController+Internal.h (hunk 1 above)
Comment (rosalyntan) [reply FirebaseAuth/Tests/Sample/Sample/MainViewController+Internal.h]: Done.

File: FirebaseAuth/Tests/Sample/Sample/MainViewController+OAuth.m (hunk 2 above)
Comment (rosalyntan) [reply FirebaseAuth/Tests/Sample/Sample/MainViewController+OAuth.m]: Done.

**Pull request merged**